master/usage/routereflector/calico-routereflector.md

```diff
@@ -0,0 +1,294 @@
+deployments.
+
+The image is currently experimental due to some key limitations discussed below.
+However, it may provide a useful framework for future development.
+
+These instructions are focused around container-based deployments that use the
+calico/node container image.
+
+For an OpenStack deployment, read [Configuring BIRD as a BGP Route Reflector](bird-rr-config).
+
+#### Known limitations
+
+-  The calico/routereflector instance will automatically peer with the Calico 
+   nodes, but it currently has no mechanism to configure peerings with non-Calico
+   BGP speakers (e.g. edge routers)
+-  There is no `calicoctl` integration or similar - it is necessary to configure 
+   data directly into the `etcd` datastore each time a new route reflector 
+   is added to the deployment.
+
+## Using the Route Reflector image
+
+### Starting a Route Reflector instance
+
+On your Route Reflector host, ensure you have [Docker v1.6](http://www.docker.com) or greater
+installed.
+
+Run the following command to start the Route Reflector container image.
+
+```
+docker run -privileged -net=host -d                                \ 
+           -e IP=<IPv4_RR>                                         \
+           [-e IP6=<IPv6_RR>]                                      \
+           -e ETCD_ENDPOINTS=<http://ETCD_IP:PORT>                 \
+           calico/routereflector:{{site.data.versions[page.version].first.components["calico/routereflector"].version}}
+```
+
+Where:
+
+-  `[]` indicates an optional parameter
+-  `<IPv4_RR>` is the IPv4 address of the RR host (the BIRD instance binds to 
+   the hosts IPv4 address)
+-  `<IPv6_RR>` is the *optional* IPv6 address of the RR host (the BIRD6 instance
+   binds to the hosts IPv6 address)
+-  `<ETCD_IP:PORT>` is the colon separated IPv4 address and port of an etcd
+   node in the etcd cluster.  A comma-separated list of endpoints may be 
+   specified.
+   
+> Note: If you require TLS/SSL enabled etcd, see the [section below](#route-reflector-with-tlsssl-etcd)
+> for details on how to start the route reflector.
+
+#### Adding the Route Reflector into etcd
+
+Add an entry in etcd for this Route Reflector.  This tells the Route Reflector
+to participate in peering, and provides enough information to allow the Route
+Reflector instances to automatically form a full BGP mesh.
+
+The configuration for the Route Reflector is stored for IPv4 at:
+
+	/calico/bgp/v1/rr_v4/<RR IPv4 address>
+	
+and IPv6 at:
+
+	/calico/bgp/v1/rr_v6/<RR IPv6 address>
+
+In all cases, the data is a JSON blob in the form:
+
+        {
+          "ip": "<IP address of BGP Peer>",
+          "cluster_id": "<Cluster ID for this RR (see notes)>"
+        }
+
+To add this entry into etcd, you could use the following commands:
+
+```
+# IPv4 entries
+curl -L http://<ETCD_IP:PORT>:2379/v2/keys/calico/bgp/v1/rr_v4/<IPv4_RR> -XPUT -d value="{\"ip\":\"<IPv4_RR>\",\"cluster_id\":\"<CLUSTER_ID>\"}"
```

Comment:
Lot of templating here. Consider laying out some sample values and just using them here:

For example, for a cluster with the following values:
- etcd running at `http://etcd.cluster:2379`
- a route reflector running at `192.168.0.1` 
- a cluster of ID `1.0.0.1`

The following command would configure this route reflector:
```
curl -L http://etcd.cluster:2379/v2/keys/calico/bgp/v1/rr_v4/192.168.0.1 -XPUT -d value="{\"ip\":\"192.168.0.1\",\"cluster_id\":\"1.0.0.1\"}"
```